tests/valgrind-check/valgrind.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+
+function print_ps {
+    set +e
+    echo "CFEngine processes:"
+    ps aux | grep [c]f-
+
+    echo "Valgrind processes:"
+    ps aux | grep [v]algrind
+    set -e
+}
+
+function stop_daemons {
+    echo "Stopping cfengine daemons"
+    pkill -f cf-serverd  || echo "Did not kill cf-serverd"
+    pkill -f cf-execd    || echo "Did not kill cf-execd"
+    pkill -f cf-monitord || echo "Did not kill cf-monitord"
+}
+
+function no_errors {
+    set +e
+    grep -i "error" $1
+    grep -i "error" $1 && exit 1
+    set -e
+}
+
+function check_daemon_output {
+    echo "Examining $1:"
+    no_errors $1
+}
+
+function check_valgrind_output {
+    set -e
+    if [ ! -f "$1" ]; then
+        echo "$1 does not exists!"
+        exit 1
+    fi
+    echo "Looking for problems in $1:"
+    grep -i "ERROR SUMMARY: 0 errors" "$1"
+    cat $1 | sed -e "/ 0 errors/d" -e "/and suppressed error/d" -e "/a memory error detector/d" > filtered.txt
+    no_errors filtered.txt
+    set +e
+    grep -i "at 0x" filtered.txt
+    grep -i "at 0x" filtered.txt && exit 1
+    grep -i "by 0x" filtered.txt
+    grep -i "by 0x" filtered.txt && exit 1
+    grep -i "Failed to connect" filtered.txt
+    grep -i "Failed to connect" filtered.txt && exit 1
+    set -e
+}
+
+function check_masterfiles_and_inputs {
+    set -e
+    echo "Comparing promises.cf from inputs and masterfiles:"
+    diff /var/cfengine/inputs/promises.cf /var/cfengine/masterfiles/promises.cf
+}
+
+set -e
+set -x
+
+# Assume we are in core directory
+
+./autogen.sh --enable-debug
```

Comment:
Please add `-C`, it speeds up `./configure` in *libntech* significantly.

Reply:
Done.